x/incentives/keeper/distribute.go

```diff
@@ -344,32 +375,80 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
 
 	locksByDenomCache := make(map[string][]lockuptypes.PeriodLock)
 	totalDistributedCoins := sdk.Coins{}
+	var gaugeDistributedCoins, gaugeDistributedCoinsCL sdk.Coins
 	for _, gauge := range gauges {
-		filteredLocks := k.getDistributeToBaseLocks(ctx, gauge, locksByDenomCache)
-		// send based on synthetic lockup coins if it's distributing to synthetic lockups
-		var gaugeDistributedCoins sdk.Coins
-		var err error
-		if lockuptypes.IsSyntheticDenom(gauge.DistributeTo.Denom) {
-			gaugeDistributedCoins, err = k.distributeSyntheticInternal(ctx, gauge, filteredLocks, &distrInfo)
+		// get pool Id corresponding to the gaugeId
+		incParams := k.GetParams(ctx).DistrEpochIdentifier
+		currEpoch := k.ek.GetEpochInfo(ctx, incParams)
+
+		isCLPoolGauge, pool := k.IsValidClGauge(ctx, gauge.Id, currEpoch.Duration)
+		// only want to run this logic if the gaugeId is associated with CL PoolId
+		if isCLPoolGauge {
+			for _, coin := range gauge.Coins {
+				// since we're only doing osmo incentive we can assume there to be only 1 coin therefore coins[0]
+				incentiveRecord := cltypes.IncentiveRecord{
+					PoolId:           pool.GetId(),
+					IncentiveDenom:   coin.Denom,
+					IncentiveCreator: k.ak.GetModuleAddress(types.ModuleName),
+					RemainingAmount:  sdk.Dec(coin.Amount),
+					// calculate amount of tokens to emit per second
+					// for ex: 10000tokens to be distributed over 1day epoch will be 1000 tokens ÷ 86,400 seconds ≈ 0.01157 tokens per second (truncated)
+					EmissionRate: sdk.NewDecFromInt(coin.Amount).QuoTruncate(sdk.NewDec(int64(currEpoch.Duration.Seconds()))),
```

Comment:
I'm a bit concerned on returing `float64` here. While casting `flaot64` to `int64` is deterministic, using floats is not in general. So I would prefer if we avoided using any method that returns floats

Can we instead return milliseconds and do a manual conversion to seconds?

Reply:
can you please give me an example of a potential security risk here? should sdk.Dec be able to handle it

Reply:
Floats are non-deterministic so if someone refactors this and accidentally uses the return of `currEpoch.Duration.Seconds()` in math operations, this will lead to an app hash.

Some examples and breakdown of non-determinism cases with floats: https://shaderfun.com/2020/10/25/understanding-determinism-part-1-intro-and-floating-points/

I suggest minimizing the use of APIs and types that can lead to non-determinism